nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCP.java

```diff
@@ -83,10 +87,49 @@
             .defaultValue(ClientAuth.REQUIRED.name())
             .build();
 
+    public static final PropertyDescriptor MIN_RECV_THREAD_POOL_SIZE = new PropertyDescriptor.Builder()
+            .name("min_receiving_threads")
+            .displayName("Min Number of Receiving Message Handler Threads")
+            .description(
+                    "The minimum number of threads for handling receiving messages ready all the time. " +
+                    "Cannot be bigger than the \"Max Number of Receiving Message Handler Threads\" or \"Max Number of TCP Connections\". " +
+                    "If not set, the value of \"Max Number of Receiving Message Handler Threads\" will be used.")
+            .addValidator(StandardValidators.createLongValidator(1, 65535, true))
+            .required(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_RECV_THREAD_POOL_SIZE = new PropertyDescriptor.Builder()
+            .name("max_receiving_threads")
+            .displayName("Max Number of Receiving Message Handler Threads")
+            .description(
+                    "The maximum number of threads might be available for handling receiving messages ready all the time. " +
+                    "Cannot be bigger than the \"Max Number of TCP Connections\". " +
+                    "If not set, the value of \"Max Number of Receiving Message Handler Threads\" will be used.")
+            .addValidator(StandardValidators.createLongValidator(1, 65535, true))
+            .required(false)
+            .build();
+
+    protected static final PropertyDescriptor RECV_POOL_BYTE_BUFFERS = new PropertyDescriptor.Builder()
+            .name("pool_byte_buffers")
+            .displayName("Pool Byte Buffers")
```

Comment:
I would call it `Pool Receive Buffers` and move the property just after `Receive Buffer Size` on the UI, and then the receiver threads property.

Reply:
As for the naming, it sounds better I will apply. The order is restricted by the fact that the processor uses a mix of its own properties and the abstract parent's ones with the fashion of an extendable hook point inherited from the parent. Later on, if this change proves itself I plan to applicate it to the other Listen processors, which makes these changes move up to the parent, eliminating this inconsistency.